File: Computer architecture work/Homework 1/homework1-Rickyocon-main/src/Computer Arch HW 1/Part2/SwapAddSub.cpp
//Method to swap two variables using addition and subtraction


#include <iostream>
//library for calculating program time
#include <ctime>
//library for getting a random number 
#include <chrono>
using namespace std;


int main ()
{
    //Start timer
    auto StartTime = std::chrono::steady_clock::now();
    //initilize rand
    int rand();
    //initilize method for counter
    int counter = 0;

    //Loop for 1000 times
    for(int i=0;i<1000;i++)
    {

        //Get a random integer for x and y from 1-1000
        int x = rand()%1000;
        int y = rand()&1000;

        //print x and y values before swap
        cout << "\nBefore swapping." << endl;
        cout << "x = " << x << ", y = " << y << endl;

        //Mehtod to swap using add/sub
        x = x + y;
        y = x - y; 
        x = x - y;
        counter += 1;

        //print x and y values after swap
        cout << "After swapping." << endl;
        cout << "x = " << x << ", y = " << y << endl;


    }

    //Print how many times the loop has run using counter variable
    cout << "The loop has run "<<counter<<" times"<<endl; 
    //End timer
    auto EndTime = std::chrono::steady_clock::now();
    //Calculate elapsed time by subtracting endtime and starttime in miliseconds
    double elapsed_time = double (std::chrono::duration_cast <std::chrono::milliseconds> (EndTime - StartTime).count());
    //print how long the program took
    cout <<"This program took "<<elapsed_time<<" ms"<<endl;
    return 0;

}

File: Computer architecture work/Homework 1/homework1-Rickyocon-main/src/Computer Arch HW 1/Part2/SwapAddSub.java
// Java program to swap two variables
// Using a addition and subtraction

public class SwapAddSub 
{

	public static void main(String[] args)
	{    
     
        //Start timer
        long StartTime = System.nanoTime(); 
       
        //initilize counter
        int counter = 0;

        //loop 1000 times
        for(int i = 1;i<=1000;i++)
        {
            //Get random value for x and y
            double x = (Math.random()*1000);
            double y = (Math.random()*1000);

            //Print values of x and y before swap
            System.out.println("Before Swap");
		    System.out.println("x = " + x);
		    System.out.println("y = " + y);

            //Swap using add/sub
		    x = x + y;
            y = x - y;
            x = x - y;
            counter += 1;

            //print values of x and y after swaping
		    System.out.println("After swap");
		    System.out.println("x = " + x);
		    System.out.println("y = " + y);
            //print how many times the loop has run
            System.out.println("The loop has run " +counter +" times");

        }

        //end timer and calculate elapsed time
        long duration = (System.nanoTime() - StartTime)/1000000;  //Calculates time in mili seconds
        //Print duration
        System.out.println("The program took " +duration + " ms");

	}
}
